inst/include/dust/interface.hpp

```diff
@@ -506,19 +506,59 @@ cpp11::sexp dust_compare_data(SEXP ptr) {
   return ret_r;
 }
 
+template <typename T, typename std::enable_if<!std::is_same<dust::no_data, typename T::data_t>::value, int>::type = 0>
+cpp11::sexp dust_filter(SEXP ptr, bool save_history) {
+  using namespace cpp11::literals;
+  Dust<T> *obj = cpp11::as_cpp<cpp11::external_pointer<Dust<T>>>(ptr).get();
+  cpp11::writable::doubles log_likelihood(obj->filter(save_history));
+  cpp11::sexp history;
+  if (save_history) {
+    cpp11::writable::doubles history_data(obj->filter_history().size());
+    obj->filter_history().history(REAL(history_data));
```

Comment:
Will this work with `real_t = float`?

Reply:
yes; `REAL(history_data)` is `double *` (that's needed on the R side) and that will go into the `copy_n` with arguments `std::vector<float>::const_iterator`, `size_t`, `double*` which the compiler will handle sensibly